src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexes/Models/Field.Customization.cs

```diff
@@ -45,12 +60,171 @@ public Field(string name, DataType dataType, AnalyzerName analyzerName)
             IsSearchable = true;
         }
 
+        /// <summary>
+        /// Initializes a new complex Field with required arguments.
+        /// </summary>
+        /// <param name="name">The name of the complex field.</param>
+        /// <param name="dataType">The data type of the field. Must be a complex type.</param>
+        /// <param name="fields">The sub-fields that comprise the complex type. They can be leaf or complex fields themselves.</param>
+        /// <exception cref="System.ArgumentException">Thrown if <c>dataType</c> is not a complex type.</exception>
+        /// <exception cref="System.ArgumentNullException">Thrown if <c>fields</c> is null.</exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">Thrown if <c>fields</c> is empty.</exception>
+        public Field(string name, DataType dataType, IList<Field> fields) : this()
+        {
+            Throw.IfArgument(!dataType.IsComplex(), nameof(dataType), "Cannot create a complex field of a non-complex type.");
+            Throw.IfArgumentNull(fields, nameof(fields));
+            Throw.IfArgumentOutOfRange(fields.Count < 1, nameof(fields), "Cannot create a complex field without sub-fields.");
```

Comment:
very small nit (and subjective too): The name of this function doesnt seem to accurately explain what it does. It doesn't really "throw if argument out of range", it "throw's argument out of range IF parameter is true". The description of the function explains what it does, but by looking at its usage here I spent some time wondering how it was able to know if something was out of range.